src/components/cesium_map/cesium_UI.js

```diff
@@ -477,8 +485,11 @@ class CesiumMap extends React.Component {
     // clear all element in cesium
     searchFields = nextProps.newSearchFields;
     clearBoundingBox(true);
-    // update the point layer
-    this.updatePrimitive(viewer.currentView.latitude, viewer.currentView.longitude);
+    // // update the point layer only when number of points differ
+    if (viewer && nextProps.numFound !== prevNumFound) {
```

Comment:
Handle the case when viewer is null (before the await operation is done) and only update the points when the number of points differ to the previous number of points fetched.